File: src/main/java/org/example/toworkspring/models/Track.java
package org.example.toworkspring.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.Type;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;

@Getter
@Setter
@Entity
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@Table(name = "track", schema = "public")
public class Track {
    @Id
    @Column(name = "id", nullable = false)
    @Type(type = "org.hibernate.type.LongType")
    private Long id;

    @NotNull
    @Column(name = "name", nullable = false)
    @Type(type = "org.hibernate.type.TextType")
    private String name;

    @NotNull
    @Column(name = "quantitymodels", nullable = false)
    @Type(type = "org.hibernate.type.IntegerType")
    private Integer quantitymodels;
}

File: src/main/kotlin/org/example/toworkspring/models/Track.kt
package org.example.toworkspring.models

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.Table

@Entity
@Table(name = "track")
class Track(
    @Id
    val id: Long = 0,
    val name: String = "",
    val quantityModules: Int = 0
)